tests/validation/tests/v3_api/test_vmwarevsphere_driver.py

```diff
@@ -0,0 +1,217 @@
+            "creation": "12h",
+            "retention": "72h",
+            "type": "etcdService"},
+        "kubeApi": {
+            "alwaysPullImages": False,
+            "podSecurityPolicy": False,
+            "serviceNodePortRange": "30000-32767",
+            "type": "kubeAPIService"}},
+    "sshAgentAuth": False}
+
+vsphereConfig = {
+    "cfgparam": ["disk.enableUUID=TRUE"],
+    "cloneFrom": CLONE_FROM,
+    "cloudConfig": "#cloud-config\r\npackages:\r\n - redis-server",
+    "cloudinit": "",
+    "contentLibrary": "",
+    "cpuCount": "2",
+    "creationType": "vm",
+    "customAttribute": ["203=CustomA", "204=CustomB"],
+    "datacenter": "/RNCH-HE-FMT",
+    "datastore": "/RNCH-HE-FMT/datastore/ranch01-silo01-vm01",
+    "diskSize": "20000",
+    "folder": "/",
+    "hostsystem": "",
+    "memorySize": "6000",
+    "network": ["/RNCH-HE-FMT/network/Private Range 172.16.128.1-21"],
+    "password": "",
+    "pool": RESOURCE_POOL,
+    "sshPassword": "tcuser",
+    "sshPort": "22",
+    "sshUser": "docker",
+    "sshUserGroup": "staff",
+    "tag": [
+        "urn:vmomi:InventoryServiceTag:04ffafd0-d7de-440c-a32c-5cd98761f812:GLOBAL",
+        "urn:vmomi:InventoryServiceTag:d00f1cf2-6822-46a0-9602-679ea56efd57:GLOBAL"
+    ],
+    "type": "vmwarevsphereConfig",
+    "username": "",
+    "vappIpallocationpolicy": "",
+    "vappIpprotocol": "",
+    "vappProperty": "",
+    "vappTransport": "",
+    "vcenter": "",
+    "vcenterPort": "443",
+}
+
+
+def test_valid_environment_variables():
+    assert RANCHER_VSPHERE_USERNAME != '',  \
+        "vSphere User is required to make a cloud credential"
+    assert RANCHER_VSPHERE_PASSWORD != '', \
+        "vSphere Password is required to make a cloud credential"
+    assert RANCHER_VSPHERE_VCENTER != '', \
+        "vCenter URL is required to make a cloud credential"
+
+
+def test_create_cluster():
+    client = get_client_for_token(ADMIN_TOKEN)
+    cloud_cred = create_vsphere_credential(client)
+    nt = create_vsphere_nodetemplate(client, cloud_cred)
+    cluster = client.create_cluster(
+        name=CLUSTER_NAME,
+        rancherKubernetesEngineConfig=rke_config)
+
+    master_pool = client.create_node_pool({
+        "type": "nodetemplate",
+        "clusterId": cluster.id,
+        "controlPlane": True,
+        "etcd": True,
+        "hostnamePrefix": CLUSTER_NAME + "-master",
+        "nodeTemplateId": nt.id,
+        "quantity": 1,
+        "worker": False,
+    })
+
+    worker_pool = client.create_node_pool({
+        "type": "nodetemplate",
+        "clusterId": cluster.id,
+        "controlPlane": False,
+        "etcd": False,
+        "hostnamePrefix": CLUSTER_NAME + "-worker",
+        "nodeTemplateId": nt.id,
+        "quantity": 1,
+        "worker": True,
+    })
+
+    client.wait_success(master_pool)
+    client.wait_success(worker_pool)
+    wait_for_cluster_node_count(client, cluster, 2, 900, 5)
```

Comment:
Add some kind of asserts on the clusters to validate it's doing what is expected.